### PR TITLE
Jc/upgrade baseline

### PR DESCRIPTION
**Pull request by jmcampanini**
* upgrade baseline to 0.16
* upgrade gradle to 4.2
* upgrade git-version + processors plugins
* use consistent plugin application (`apply plugin`)
* cleaned up circle.yml to collect tests and cache dependencies

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
-
 apply plugin: 'java'
+apply plugin: 'com.palantir.git-version'
 apply plugin: 'com.palantir.baseline-config'
 apply plugin: 'com.palantir.baseline-idea'
 apply plugin: 'com.palantir.baseline-eclipse'
 apply plugin: 'com.palantir.baseline-checkstyle'
-apply plugin: 'com.palantir.baseline-findbugs'
+apply plugin: 'com.palantir.baseline-error-prone'
+apply plugin: 'org.inferred.processors'
```
Comment (qinfchen) [inline build.gradle]: this needs to go before error-prone, no?

**Review by qinfchen — APPROVED** [review]

**Comment by jmcampanini** [comment]
re-review please. at least for a sanity check... :)

**Review by qinfchen — APPROVED** [review]

**Review by CalvinFernandez — APPROVED** [review]

**Comment by jmcampanini** [comment]
@CalvinFernandez should be good to rebase #78 and remove the guava line :)

File: build.gradle [hunk 2]
```diff
@@ -47,3 +51,16 @@ group 'com.palantir.websecurity'
 version gitVersion()
 
 sourceCompatibility = 1.8
+
+task resolveConfigurations {
```
Comment (gracew) [inline build.gradle:55]: why not use the configuration resolver plugin? :'(

Comment (jmcampanini) [reply build.gradle:55]: #80 :)

**Pull request merged**